spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -763,7 +766,8 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           binding,
           (builder, binaryExpr) => builder.setLtEq(binaryExpr))
 
-      case Literal(value, dataType) if supportedDataType(dataType, allowStruct = value == null) =>
+      case Literal(value, dataType)
+          if supportedDataType(dataType, allowComplex = value == null) =>
```

Comment:
nit: would be nice to have a comment here explaining why `allowComplex = value == null`